src/test/Agent/AgentLiveProbes.test.tsx

```diff
@@ -0,0 +1,226 @@
+  recordStackTrace: true,
+  useRethrow: true,
+  methodName: 'a_method',
+  methodDescriptor: 'method_desc',
+  location: 'some_loc',
+  returnValue: 'a_value',
+  parameters: 'some_params',
+  fields: 'some_fields',
+  path: 'some_path',
+};
+
+const mockAnotherProbe: EventProbe = {
+  ...mockProbe,
+  id: 'another_id',
+  name: 'another_name',
+};
+
+const mockApplyTemplateNotification = {
+  meta: {
+    category: NotificationCategory.ProbeTemplateApplied,
+    type: mockMessageType,
+  } as MessageMeta,
+  message: {
+    template: mockProbe,
+  },
+} as NotificationMessage;
+
+const mockRemoveProbesNotification = {
+  meta: {
+    category: NotificationCategory.ProbesRemoved,
+    type: mockMessageType,
+  } as MessageMeta,
+  message: {
+    target: mockTarget,
+  },
+} as NotificationMessage;
+
+jest.spyOn(defaultServices.target, 'target').mockReturnValue(of(mockTarget));
+jest.spyOn(defaultServices.target, 'authFailure').mockReturnValue(of());
+
+jest
+  .spyOn(defaultServices.settings, 'deletionDialogsEnabledFor')
+  .mockReturnValueOnce(false) // should remove all probes when Remove All Probe is clicked
+  .mockReturnValue(true); // should show warning modal and remove all probes when confirmed
+
+jest
+  .spyOn(defaultServices.api, 'getActiveProbes')
+  .mockReturnValueOnce(of([mockProbe])) // renders correctly
+
+  .mockReturnValueOnce(of([mockProbe])) // should add a probe after receiving a notification
+  .mockReturnValueOnce(of([mockProbe, mockAnotherProbe]))
+
+  .mockReturnValue(of([mockProbe])); // All other tests
+
+jest
+  .spyOn(defaultServices.notificationChannel, 'messages')
+  .mockReturnValueOnce(of()) // renders correctly
+  .mockReturnValueOnce(of())
+  .mockReturnValueOnce(of(mockApplyTemplateNotification)) // should add a probe after receiving a notification
+  .mockReturnValueOnce(of())
+  .mockReturnValueOnce(of(mockRemoveProbesNotification))
+  .mockReturnValue(of()); // All other tests
+
+describe('<AgentLiveProbes />', () => {
+  afterEach(cleanup);
+
+  it('renders correctly', async () => {
+    let tree;
+    await act(async () => {
+      tree = renderer.create(
+        <ServiceContext.Provider value={defaultServices}>
+          <AgentLiveProbes />
+        </ServiceContext.Provider>
+      );
+    });
+    expect(tree.toJSON()).toMatchSnapshot();
+  });
+
+  it('should add a probe after receiving a notification', () => {
+    render(
+      <ServiceContext.Provider value={defaultServices}>
+        <AgentLiveProbes />
+      </ServiceContext.Provider>
+    );
+
+    const addTemplateName = screen.getByText('another_name');
+    expect(addTemplateName).toBeInTheDocument();
+    expect(addTemplateName).toBeVisible();
+  });
+
+  it('should remove a probe after receiving a notification', () => {
+    render(
+      <ServiceContext.Provider value={defaultServices}>
+        <AgentLiveProbes />
+      </ServiceContext.Provider>
+    );
+    expect(screen.queryByText('another_name')).not.toBeInTheDocument();
```

Comment:
We should check that the table is empty. Perhaps both `another_name` and `some_name`? Name of test should be `should remove all probes...`?